command-expansion-server/src/app.ts

```diff
@@ -333,6 +339,58 @@ app.post('/expand-all-activity-instances', async (req, res, next) => {
   return next();
 });
 
+app.post('/get-seqjson-for-sequence', async (req, res, next) => {
+  // Get the specified sequence + activity instance ids + commands from the latest expansion run for each activity instance (filtered on simulation dataset)
+  // get start time for each activity instance and join with activity instance command data
+  // Create sequence object with all the commands and return the seqjson
+  const context: Context = res.locals.context;
+
+  const seqId = req.body.input.seqId as string;
+  const simulationDatasetId = req.body.input.simulationDatasetId as number;
+
+  const { rows } = await db.query<{
+    metadata: Record<string, unknown>;
+    commands: ReturnType<Command['toSeqJson']>,
+    activity_instance_id: number,
+    errors: string,
+  }>(`
+    select distinct on (activity_instance_commands.expansion_run_id) sequence.metadata, activity_instance_commands.commands, activity_instance_commands.activity_instance_id, activity_instance_commands.errors
+    from sequence
+      join sequence_to_activity_instance on sequence.seq_id = sequence_to_activity_instance.seq_id and sequence.simulation_dataset_id = sequence_to_activity_instance.simulation_dataset_id
+      join activity_instance_commands on activity_instance_commands.activity_instance_id = sequence_to_activity_instance.activity_instance_id
+      join expansion_run on activity_instance_commands.expansion_run_id = expansion_run.id
+        where expansion_run.simulation_dataset_id = $1 and sequence.seq_id = $2
+    order by activity_instance_commands.expansion_run_id desc;
+    `,
+      [simulationDatasetId, seqId],
+  );
+
+
+  const activityInstances = await context.simulatedActivityInstanceDataLoader.loadMany(rows.map(row => ({ simulationDatasetId, activityInstanceId: row.activity_instance_id})));
+  const activityInstanceLoadErrors = activityInstances.filter(ai => ai instanceof Error);
+  if (activityInstanceLoadErrors.length > 0) {
+    res.status(500).json({
+      error: 'Error loading activity instances',
+      activityInstanceLoadErrors,
+    });
+    return next();
+  }
+  const seqMetadata = rows[0].metadata;
+  const sortedActivityInstances = (activityInstances as Exclude<typeof activityInstances[number], Error>[])
+      .map(ai => ai[0])
+      .sort((a, b) => Temporal.Duration.from(parseInterval(a.start_offset).toISOString()).nanoseconds - Temporal.Duration.from(parseInterval(b.start_offset).toISOString()).nanoseconds)
+
+  const sortedActivityInstancesWithCommands: (SimulatedActivityInstance<{[p: string]: unknown}, {[p: string]: unknown}> & { commands: ReturnType<Command['toSeqJson']> })[] = sortedActivityInstances.map(ai => ({ ...ai, commands: assertDefined(rows.find(row => row.activity_instance_id === ai.id)).commands }));
+
+  // TODO: Come up with a way to make this configurable and load that configured seq builder if present
+  const seqBuilder = defaultSeqBuilder;
```

Comment:
Good idea to make this configurable

Reply:
Discussed with Pat and Matt and while this is useful, until there is a request for doing more complicated things when building the sequence, we won't be implementing this. And in fact, this might force that conversation of whether we should be doing sequence building at all and encourage them to just build a couple hundred line python script to pull from the api and do the building themselves.